mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java

```diff
@@ -875,18 +887,22 @@ private static MCRObject retrieveMCRObject(String idString) throws MCRRestAPIExc
         }
     }
 
-    private static MCRDerivate retrieveMCRDerivate(MCRObjectID parentObjId, String derIDString)
+    //TODO move into generic package for v1 and v2 REST-API
+    public static MCRObjectID retrieveMCRDerivateID(MCRObjectID parentObjId, String derIDString)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'retrieveMCRDerivateID(MCRObjectID, String)' has an NPath complexity of 8100](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)